src/components/atoms/InputField/InputField.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { forwardRef } from "react";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+
+import { isTruthy } from "utils/types";
+
+import "./InputField.scss";
+
+export interface InputFieldProps extends React.HTMLProps<HTMLInputElement> {
+  iconStart?: IconProp | JSX.Element;
+  iconEnd?: IconProp | JSX.Element;
+}
```

Comment:
Also, I think we need to add typing to support passing through the `extraProps` here.

I believe this is called 'variadic' paramaters.

**Further Reading:**

- https://stackoverflow.com/questions/12699058/typescript-types-for-variadic-functions
- https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#variadic-tuple-types
- https://www.infoq.com/news/2020/08/typescript-4-variadic-tuple-type/

---

Looking at the types we can pass to the `input` itself:

![Screen Shot 2021-04-12 at 1 08 42 pm](https://user-images.githubusercontent.com/753891/114335515-4a5ae980-9b90-11eb-8855-080f289e9ecb.png)

It's defined in `node_modules/@types/react/index.d.ts:3021` as:

```typescript
React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>;
```

With the most relevant aspect seeming to be defined in `node_modules/@types/react/index.d.ts:2067`

```typescript
interface InputHTMLAttributes<T> extends HTMLAttributes<T> {
  // ..snip..
}
```

So I believe the type we would want to use for this would be: `React.InputHTMLAttributes<HTMLInputElement>`

I'm not sure if that is possible using the variadic parameters.. so perhaps we would just need to make it an explicitly named parameter (eg. `extraInputProps`) in the types/etc as well.